packages/calcite-components/src/components/autocomplete/autocomplete.tsx

```diff
@@ -444,9 +446,18 @@ export class Autocomplete
       this.reposition(true);
     }
 
+    if (changes.has("inputValue") && (this.hasUpdated || this.inputValue)) {
+      this.inputValueMatchPattern =
+        this.inputValue && new RegExp(`(${escapeRegExp(this.inputValue)})`, "i");
+      this.updateItems();
+      this.updateGroups();
+      return;
```

Comment:
For maintainability, could you use flags and call the corresponding methods at the end of `willUpdate` instead of returning early? This will ensure any new checks added won’t need to account for early returns.

Reply:
done!